applications/sintering/include/pf-applications/sintering/operator_advection.h

```diff
@@ -92,6 +92,61 @@ namespace Sintering
       src.zero_out_ghost_values();
     }
 
+    bool
+    check_courant(const AdvectionMechanism<dim, Number, VectorizedArrayType>
+                    &          advection_mechanism,
+                  const double dt) const
+    {
+      MyScope scope(this->timer,
+                    "advection_op::check_courant",
+                    this->do_timing);
+
+      FECellIntegrator<dim, 1, Number, VectorizedArrayType> phi(
+        this->matrix_free, this->dof_index);
+
+      VectorizedArrayType cell_diameters;
+      VectorizedArrayType zeros(0.0);
+      VectorizedArrayType ones(1.0);
+
+      std::pair<unsigned int, unsigned int> range{
+        0, this->matrix_free.n_cell_batches()};
+
+      for (auto cell = range.first; cell < range.second; ++cell)
+        {
+          phi.reinit(cell);
+
+          // Reinit advection data for the current cells batch
+          advection_mechanism.reinit(cell);
+
+          // Get cell diameters
+          for (unsigned int i = 0; i < VectorizedArrayType::size(); ++i)
+            {
+              const auto icell  = this->matrix_free.get_cell_iterator(cell, i);
+              cell_diameters[i] = icell->diameter();
+            }
```

Comment:
Maybe we should precompute this? Not needed in this PR.

Reply:
Good idea, makes sense to do it after each refinement then.